modules/core/src/v2/coins/sol.ts

```
@@ -288,4 +289,35 @@ export class Sol extends BaseCoin {
 
     return explainedTransaction as SolTransactionExplanation;
   }
+
+  /** @inheritDoc */
+  supportsDerivationKeypair(): boolean {
+    return true;
+  }
+
+  /** @inheritDoc */
+  deriveKeypair(params: DeriveKeypairOptions): KeyPair | undefined {
+    if (_.isNil(params.addressDerivationPrv)) {
+      throw new Error('addressDerivationPrv is missing');
+    }
+    try {
+      const rootKeypair = new accountLib.Sol.KeyPair({ prv: params.addressDerivationPrv });
+      const derivedKeys = rootKeypair.deriveHardened(`m/0'/0'/0'/${params.index}'`);
+      if (_.isNil(derivedKeys.prv)) {
+        throw new Error('Key derivation failed - missing derived prv key');
+      }
+      // Necesary step because the derivedKeys pub is not a base58 key
+      const derivedBase58KeyPair = new accountLib.Sol.KeyPair({ prv: derivedKeys.prv }).getKeys();
+      if (!_.isString(derivedBase58KeyPair.prv)) {
+        throw new Error('Key derivation failed - missing derived base58 prv key');
+      }
+
+      return {
+        prv: derivedBase58KeyPair.prv,
+        pub: derivedBase58KeyPair.pub,
+      };
+    } catch (e) {
+      return undefined;
+    }
```

no need to catch imo, let's surface the error back to the user